lib/param_parsing/parameters_test.go

```diff
@@ -91,17 +92,76 @@ func TestGetParameters_dry_run_wont_download_anything(t *testing.T) {
 	})
 }
 
-func TestGetParameters_check_config_precedence(t *testing.T) {
+func writeTestFile(t *testing.T, basePath string, fileName string, fileContent string) {
+	fullPath := filepath.Join(basePath, fileName)
+	if err := os.WriteFile(fullPath, []byte(fileContent), 0666); err != nil {
```

Comment:
The `0666` is too permissive IMHO.

Reply:
I can change it for sure. Only bearing in mind that it's just for the tests and the file is created in the git tree, so if someone is using VSCode with a dev container which is running as root and the test fails to clear up the files, removing them from the host OS as a standard user would cause a little friction

Reply:
I might be running a bit "too much", but write permission for group and others is a potential security breach that allows arbitrary actors to write to a file even during a relatively short period of time 🤷🏻 My take is always use what's defined by `umask` (incl user's override for system default)

Reply:
I'm happy to make the change to make it more restrictive.
Though, from what I read in the docs for the functions:
 * WriteFile (https://pkg.go.dev/os#WriteFile) - indicates that, if the file does not exist, it creates the file with the "required permission (before umask)"

However, `io.Create`, the function that was also used in tfswitche's files.go states:
`if the file does not exist, it is created with mode 0666 (before umask)`
and this does appear to take the umask into account.

Reply:
> Though, from what I read in the docs for the functions:

Ah, my bad, I just re-read the docu and tested it and it indeed takes my custom umask value into account.
So the `perm` looks to be a fallback, though my understanding of "sane" differs =)
I'd appreciate if you changed it to `0600` please as it's apparently us to look after security when using these bits in Go 🤪